torch_np/_ndarray.py

```diff
@@ -3,11 +3,40 @@
 import torch
 
 from . import _util
+from . import _helpers
 from . import _dtypes
 
 NoValue = None
 newaxis = None
 
+
+def axis_out_keepdims_wrapper(func):
+    """`func` accepts an array-like as a 1st arg, returns a tensor.
+
+    This decorator implements the generic handling of axis, out and keepdims
+    arguments for reduction functions.
+    """
+    # XXX: move this out of _ndarray.py (circular imports)
+    @functools.wraps(func)
+    def wrapped(a, axis=None, out=None, keepdims=NoValue, *args, **kwds):
+        arr = asarray(a)
+        axis = _helpers.standardize_axis_arg(axis, arr.ndim)
+
+        arr1 = arr
```

Comment:
why?

Reply:
So you have a preference between

```
add1 = arr
if axis == ():
     arr1 = ...
```

vs

```
if axis == ():
    arr1 = ...
else:
    arr1 = arr
```
?

Reply:
My point is, why not doing this using the `arr` name itself. Then simply do `arr = arr.reshape(newshape)`

Reply:
OK, done.